### PR TITLE
May june July 2024

### PR DESCRIPTION
**Pull request by SethBarberee**
The behemoth of git rebasing is fixed.

### CODE REVIEW

File: src/math.c [hunk 1]
```diff
@@ -159,3 +157,165 @@ u24_8 u24_8_mul(u24_8 x, u24_8 y) {
...
+            r5 = r5 + s24_8_mul(r5,r4) * 2;
+            r6 = s24_8_mul(r6,r4);
+        }
+    }
+    return r5;
+}
+
+void sub_800A020(s32 *param_1, u32 param_2)
+{
+  register u32 temp asm("r4");
```
Comment (AnonymousRandomPerson) [inline src/math.c:291]: Add `NONMATCHING`.

File: src/code_8097670.c [hunk 2]
```diff
@@ -201,4 +210,203 @@ bool8 sub_8097900(s16 pokeIndex)
...
+                if ((move->moveFlags & 1) != 0) {
+                    gUnknown_203B494->unk8C[move->id / 32] |= 1 << (move->id & 0x1f);
+                }
+            }
+        }
+  }
+    
+  for(index = 0; index < MONSTER_MAX; index++)
+  {
+    if (index != MONSTER_NONE)
```
Comment (DizzyEggg) [inline src/code_8097670.c:261]: This look terrible lol. Are you sure this couldn't be changed to something else? `continue` maybe?

Comment (SethBarberee) [reply src/code_8097670.c:261]: Haven't looked at it in a while but here's the scratch link:

https://decomp.me/scratch/wH4qc

Comment (mateon1) [reply src/code_8097670.c:261]: At the very least the horrible if() if() if() dance can be converted to a single if (.. && .. && .. && ..)

Comment (mateon1) [reply src/code_8097670.c:261]: Here's a cleaned up scratch, had a go at the NONMATCHING but couldn't get it to go away https://decomp.me/scratch/WyTn5

**Review by AnonymousRandomPerson — APPROVED** [review]

**Pull request merged**